airflow/executors/executor_loader.py

```diff
@@ -69,56 +71,181 @@ class ExecutorLoader:
+                # executors by providing an alias along with the existing core airflow executor alias
+                # (e.g. my_local_exec_alias:LocalExecutor). Allowing this makes things unnecessarily
+                # complicated. Multiple Executors of the same type will be supported by a future multitenancy
+                # AIP.
+                # The module component should always be a module or plugin path.
+                if not split_name[1] or split_name[1] in CORE_EXECUTOR_NAMES:
+                    raise AirflowConfigException(
+                        f"Incorrectly formatted executor configuration: {name}\n"
+                        "second portion of an executor configuration must be a module path"
+                    )
+                else:
+                    executor_names.append(ExecutorName(alias=split_name[0], module_path=split_name[1]))
+            else:
+                raise AirflowConfigException(f"Incorrectly formatted executor configuration: {name}")
+
+        # As of now, we do not allow duplicate executors.
+        # Add all module paths/plugin names to a set, since the actual code is what is unique
+        unique_modules = set([exec_name.module_path for exec_name in executor_names])
+        if len(unique_modules) < len(executor_names):
+            msg = (
+                "At least one executor was configured twice. Duplicate executors is not yet supported. "
+                "Please check your configuration again to correct the issue."
+            )
+            raise AirflowConfigException(msg)
+
+        # Populate some mappings for fast future lookups
+        for executor_name in executor_names:
+            # Executors will not always have aliases
+            if executor_name.alias:
+                _alias_to_executors[executor_name.alias] = executor_name
+            # All executors will have a module path
+            _module_to_executors[executor_name.module_path] = executor_name
+            # Cache the executor names, so the logic of this method only runs once
+            _executor_names.append(executor_name)
+
+        return executor_names
+
+    @classmethod
+    def get_default_executor_name(cls) -> ExecutorName:
+        """Return the default executor name from Airflow configuration.
+
+        :return: executor name from Airflow configuration
+        """
+        # The default executor is the first configured executor in the list
+        return cls._get_executor_names()[0]
 
     @classmethod
     def get_default_executor(cls) -> BaseExecutor:
         """Create a new instance of the configured executor if none exists and returns it."""
-        if cls._default_executor is not None:
-            return cls._default_executor
+        default_executor = cls.load_executor(cls.get_default_executor_name())
+
+        return default_executor
+
+    @classmethod
+    def init_executors(cls) -> list[BaseExecutor]:
```

Comment:
Not used by any code here. Will be used by the scheduler and backfill job runners (or more precisely the jobs within).